pkg/apis/apps/types.go

```diff
@@ -119,6 +174,21 @@ type StatefulSetStatus struct {
 
 	// Replicas is the number of actual replicas.
 	Replicas int32
+
+	// readyReplicas is the number of replicas with a Ready Condition.
+	ReadyReplicas int32
+
+	// currentReplicas is the number of replicas at the CurrentRevision
+	CurrentReplicas int32
+
+	// updatedReplicas is the number of replicas
+	UpdatedReplicas int32
+
+	// currentRevision is the current revision of the StatefulSet
+	CurrentRevision string
+
+	// updateRevision is the revision the StatefulSet is attempting to roll out
+	UpdateRevision string
```

Comment:
I find the comments in proposal very clear and useful, copy them here? https://github.com/kubernetes/community/blob/master/contributors/design-proposals/statefulset-update.md#api-objects